pkgs/os-specific/linux/kernel-headers/default.nix

```diff
@@ -22,16 +20,35 @@ let
 
     extraIncludeDirs = lib.optional stdenvNoCC.hostPlatform.isPowerPC ["ppc"];
 
-    # "patches" array defaults to 'null' to avoid changing hash
-    # and causing mass rebuild
     inherit patches;
 
-    buildPhase = ''
-      make mrproper headers_check SHELL=bash
+    hardeningDisable = lib.optional stdenvNoCC.buildPlatform.isDarwin "format";
+
+    makeFlags = [
+      "SHELL=bash"
+      # Avoid use of runtime build->host compilers for checks
+      "cc-version:=9999"
+      "cc-fullversion:=999999"
+      # `$(..)` expanded by make alone
+      "HOSTCC:=$(BUILD_CC)"
+      "HOSTCXX:=$(BUILD_CXX)"
+    ];
+
+    # Skip clean on darwin, case-sensitivity issues.
+    buildPhase = lib.optionalString (!stdenvNoCC.buildPlatform.isDarwin) ''
+      make mrproper $makeFlags
+    '' + ''
+      make headers_install $makeFlags
+    '';
+
+    # doCheck = !stdenvNoCC.buildPlatform.isDarwin;
```

Comment:
What's with this?

Reply:
I pulled out in https://github.com/NixOS/nixpkgs/pull/46574, let's review there?